### PR TITLE
[stable-1.71] Bump version of crates-io due to unintentional semver-breaking change

### LINKED ISSUE #12354: crates-io 0.36.1 breaks semver
### Problem

At least `rust_version` got added to `NewCrate`.

### Steps

Open 0.36.0 and 0.36.1 and compare.

### Possible Solution(s)

yank 0.36.1 and release it as 0.37.0

### Notes

_No response_

### Version

_No response_

### PR DESCRIPTION
**Pull request by arlosi**
r? @weihanglo 

Fixes #12354

### CODE REVIEW

File: Cargo.lock [hunk 1]
```diff
@@ -2472,9 +2472,9 @@ dependencies = [
 
 [[package]]
 name = "proc-macro2"
-version = "1.0.56"
+version = "1.0.64"
```
Comment (weihanglo) [inline Cargo.lock:2475]: Why does this change? 🤔

**Review by weihanglo — COMMENTED** [review]
Thanks!

We also need to do a similar one against `rust-1.72.0` maybe?

File: Cargo.lock (hunk 1 above)
Comment (arlosi) [reply Cargo.lock:2475]: The nightly CI build was broken without it.

Comment (arlosi) [reply Cargo.lock:2475]: See https://github.com/rust-lang/rust/issues/113152

Comment (weihanglo) [reply Cargo.lock:2475]: I see. Thanks!

**Review by weihanglo — APPROVED** [review]

**Pull request merged**